src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -432,24 +416,32 @@ object ScalafixPlugin extends AutoPlugin {
         (ScalafixConfig / externalDependencyClasspath).value
           .flatMap(_.get(moduleID.key))
 
-      if (shellArgs.rules.isEmpty && shellArgs.extra == List("--help")) {
-        scalafixHelp
+      val allResolvers =
+        ((ThisBuild / scalafixResolvers).value ++ (ThisBuild / scalafixSbtResolversAsCoursierRepositories).value).distinct
+      val (shell, baseInterface) = scalafixArgsFromShell(
+        shellArgs,
+        projectDepsExternal,
+        projectDepsInternal,
+        scalafixScalaBinaryVersion.value,
+        scalafixDependencies.value,
+        allResolvers,
+        (ThisBuild / scalafixCallback).value,
+        scalafixInterfaceCache.value
+      )
+
+      val informationalInvocation = shellArgs.extra
+        .intersect(List("--help", "-h", "--version", "-v"))
```

Comment:
this removes the need for triggering compilation when `--version` or `-v` is requested